Services/Database/interfaces/interface.ilDBInterface.php

```diff
@@ -293,4 +295,12 @@ public function buildAtomQuery(): ilAtomQuery;
     public function groupConcat(string $a_field_name, string $a_seperator = ",", ?string $a_order = null): string;
 
     public function cast(string $a_field_name, string $a_dest_type): string;
+
+    public function addForeignKey(string $foreign_key_name, string $field_name, string $table_name, string $reference_field_name, string $reference_table, ?string $on_update = null, ?string $on_delete = null): bool;
```

Comment:
IMO we should avoid to use primitive strings for the `$on_update` and `$on_delete`. We should use a collection (an array would be OK IMO) of `ReferenceOption` types. A `ReferenceOption` could encapsulate a type (`ON UPDATE` or `DELETE`) and the `ReferentialAction` (the cases you used in `ilForeignKeyConstraints`).